addons/stock_account/wizard/stock_valuation_layer_revaluation.py

```diff
@@ -21,7 +21,7 @@ def default_get(self, default_fields):
                 raise UserError(_("You cannot revalue a product with a standard cost method."))
             if product.quantity_svl <= 0:
                 raise UserError(_("You cannot revalue a product with an empty or negative stock."))
-            if 'account_journal_id' not in res and 'account_journal_id' in default_fields:
+            if 'account_journal_id' not in res and 'account_journal_id' in default_fields and self.property_valuation == 'real_time':
```

Comment:
`self` isn't instantiate yet, then `self.property_valuation` will be empty (=> use `product.categ_id.property_valuation` instead). 
Also can you move the line `accounts = product.product_tmpl_id.get_product_accounts()` inside the this if ?